### PR TITLE
Hg 618 hacking poll daddy back together again

### PR DESCRIPTION
**Pull request by lizlux**
PollDaddy relies on global variables that don't play nicely with single page apps. In lieu of an elegant solution, we've decided to put a bandaid on it to get it working quickly in mercury.

### CODE REVIEW

File: front/scripts/main/views/ArticleView.ts [hunk 1]
```diff
@@ -230,6 +231,31 @@ App.ArticleView = Em.View.extend(App.AdsMixin, App.ViewportMixin, {
...
 	},
 
+	/**
+	 * This is a hack to make PollDaddy work (HG-618)
+	 * @see http://static.polldaddy.com/p/8791040.js
+	 */
+	handlePollDaddy: function (): void {
+		var $polls = this.$('script[src*=polldaddy]');
+
+		$polls.each((index: number, script: HTMLScriptElement) => {
```
Comment (rogatty) [inline front/scripts/main/views/ArticleView.ts]: `: void`

File: front/scripts/main/views/ArticleView.ts [hunk 2]
```diff
@@ -230,6 +231,31 @@ App.ArticleView = Em.View.extend(App.AdsMixin, App.ViewportMixin, {
...
+			// extract ID from script src
+			var idRegEx: RegExp = /(\d+)\.js$/,
+				id: number = parseInt(script.src.match(idRegEx)[1], 10),
+				html;
+
+			// avoid PollDaddy's document.write on subsequent article loads
+			if (!this.$('#PDI_container' + id).length) {
+				html = '<a name="pd_a_' + id + '" style="display: inline; padding: 0px; margin: 0px;"></a>' +
+					'<div class="PDS_Poll" id="PDI_container' + id + '"></div>';
+				$(script).after(html)
```
Comment (rogatty) [inline front/scripts/main/views/ArticleView.ts]: Missing `;`

**Comment by rogatty** [comment]
```
PhantomJS 1.9.8 (Linux) Application View isMWContent returns correct value FAILED
	Died on test #1     at /var/lib/jenkins/workspace/Mercury-Front/node_modules/qunitjs/qunit/qunit.js:263
	    at /var/lib/jenkins/workspace/Mercury-Front/www/front/vendor/sinon-qunit/lib/sinon-qunit.js:27
	    at test (/var/lib/jenkins/workspace/Mercury-Front/front/vendor/ember-qunit/ember-qunit.js:252)
	    at /var/lib/jenkins/workspace/Mercury-Front/test/specs/front/scripts/main/views/ApplicationView.js:59: 'undefined' is not a function (evaluating 'appViewMock.isMWContent(testCase.target)')
	TypeError: 'undefined' is not a function (evaluating 'appViewMock.isMWContent(testCase.target)')
```

**Comment by lizlux** [comment]
Thanks @rogatty, fixed.

**Comment by hakubo** [comment]
Wow, wow haven't we just added https://qualaroo.com/ ?
Do we really have to have 2 services to do almost the same?

**Comment by hakubo** [comment]
Ok now I see the ticket...
We are doomed :(

**Comment by lizlux** [comment]
@hakubo, yeaaah.

File: front/scripts/main/views/ArticleView.ts [hunk 3]
```diff
@@ -230,6 +231,31 @@ App.ArticleView = Em.View.extend(App.AdsMixin, App.ViewportMixin, {
...
+	 * This is a hack to make PollDaddy work (HG-618)
+	 * @see http://static.polldaddy.com/p/8791040.js
+	 */
+	handlePollDaddy: function (): void {
+		var $polls = this.$('script[src*=polldaddy]');
+
+		$polls.each((index: number, script: HTMLScriptElement): void => {
+			// extract ID from script src
+			var idRegEx: RegExp = /(\d+)\.js$/,
+				id: string = script.src.match(idRegEx)[1],
```
Comment (rogatty) [inline front/scripts/main/views/ArticleView.ts]: Should we return if `id === undefined`?

File: front/scripts/main/views/ArticleView.ts [hunk 4]
```diff
@@ -230,6 +231,31 @@ App.ArticleView = Em.View.extend(App.AdsMixin, App.ViewportMixin, {
...
+
+			// avoid PollDaddy's document.write on subsequent article loads
+			if (!this.$('#PDI_container' + id).length) {
+				html = '<a name="pd_a_' + id + '" style="display: inline; padding: 0px; margin: 0px;"></a>' +
+					'<div class="PDS_Poll" id="PDI_container' + id + '"></div>';
+				$(script).after(html);
+			}
+
+			// init PollDaddy
+			window['PDV_go' + id]();
```
Comment (rogatty) [inline front/scripts/main/views/ArticleView.ts]: Maybe we should check if this is an actual function.

**Comment by kenkouot** [comment]
LGTM, thanks for handling this. Also @hakubo yes :(

File: front/scripts/main/views/ArticleView.ts (hunk 3 above)
Comment (lizlux) [inline front/scripts/main/views/ArticleView.ts]: fixed in https://github.com/Wikia/mercury/pull/823

**Pull request merged**